### PR TITLE
feat(web): add schema menu

### PR DESCRIPTION
**Pull request by issmail-basel**
# Overview

## What I've done
Added a new molecule component for showing a list of models, and the ability to create new models.
Also added typescript types for both `model` and `field`.

## How I tested
Tested in the acceptance test

## Screenshot
schema

## Which point I want you to review particularly
Syntax issues

### CODE REVIEW

File: web/src/components/molecules/Schema/types.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+export type Model = {
+  id: string;
+  name: string;
+  description: string;
```
Comment (KaWaite) [inline web/src/components/molecules/Schema/types.ts:4]: Isn't description optional? (In your modalCreationModel PR the form doesn't make description required)

File: web/src/components/molecules/Schema/types.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  key: string;
```
Comment (KaWaite) [inline web/src/components/molecules/Schema/types.ts:5]: Are these optional? In the modelcreationmodal PR they aren't required, but start with empty string values, so I am a little unsure if that is on purpose and/or desired.

File: web/src/components/molecules/Schema/SchemaMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  selectModel: (modelId: string) => void;
+}
+
+const SchemaMenu: React.FC<Props> = ({
+  defaultSelectedKeys,
+  models,
+  handleModalOpen,
+  selectModel,
+}) => {
+  const onClick = (e: any) => {
```
Comment (KaWaite) [inline web/src/components/molecules/Schema/SchemaMenu.tsx:26]: Can this be typed better to whatever the Menu component's onClick event is?

File: web/src/components/molecules/Schema/SchemaMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  };
+
+  return (
+    <SchemaStyledMenu>
+      <SchemaStyledTitle>Schema</SchemaStyledTitle>
+      <SchemaAction>
+        <SchemaStyledMenuTitle>Models</SchemaStyledMenuTitle>
+        <SchemaAddButton
+          onClick={handleModalOpen}
+          color="#1890FF"
```
Comment (KaWaite) [inline web/src/components/molecules/Schema/SchemaMenu.tsx:37]: You have the color set here and below in the styled components. Pick one and remove the other

File: web/src/components/molecules/Schema/SchemaMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    color: #1890ff;
+  }
+`;
+
+const SchemaStyledMenuTitle = styled.h1`
+  margin: 0;
+  font-weight: 400;
+  font-size: 14px;
+  line-height: 22px;
+  color: rgba(0, 0, 0, 0.45);
```
Comment (KaWaite) [inline web/src/components/molecules/Schema/SchemaMenu.tsx:74]: Can you use hex here since you are using hex everywhere else? Just to keep consistent

**Review by KaWaite — COMMENTED** [review]
Couple comments

File: web/src/components/molecules/Schema/types.ts (hunk 2 above)
Comment (issmail-basel) [reply web/src/components/molecules/Schema/types.ts:5]: @KaWaite , I think key is required, and it is a backend edit that should be done as well

**Review by KaWaite — APPROVED** [review]

**Pull request merged**